app/components/ImageSearch.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import { extractImagesFromPDF, getImageEmbedding, getTextEmbedding, searchEmbeddings } from '../utils/imageSearchHelpers';
+
+interface ImageSearchProps {
+  pdfUrl: string;
+}
+
+const ImageSearch: React.FC<ImageSearchProps> = ({ pdfUrl }) => {
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [results, setResults] = useState<{ similarity: number; index: number }[]>([]);
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+
+  const handleSearch = async () => {
+    setIsSearching(true);
+
+    try {
+      // Step 1: Extract images from the PDF using the helper function
+      const images = await extractImagesFromPDF(pdfUrl);
+
+      // Step 2: Generate embeddings for each image
+      const imageEmbeddings = await Promise.all(images.map(async (img) => getImageEmbedding(img)));
+
+      // Step 3: Generate text query embedding
+      const queryEmbedding = await getTextEmbedding(searchQuery);
+
+      // Step 4: Perform search by comparing the query embedding with image embeddings
+      const matchingResults = searchEmbeddings(queryEmbedding, imageEmbeddings);
```

Comment:
Nice, good clarity. Have you thought about doing this on the server side?

Reply:
Thank you! I did actually plan on doing it on the server side in Node.js because I thought it would work just as well if not better, but I decided to keep it consistent with all the other components of the web app and wrote it in TypeScript on the frontend.